docs/quality_of_life/stack_frame_utils.md

```diff
@@ -0,0 +1,33 @@
+---
+icon: octicons/stack-24
+search:
+    boost: 4
+---
+# :octicons-stack-24: Stack Frame Utils
+Function calls in a binary executable are made according to a system calling convention. One constant in these conventions is the use of a stack frame to store the return addresses to resume at the end of the function.
+
+Different architectures have slightly different ways to retrieve the return address (for example, in AArch64, the latest return address is stored in `x30`, the Link Register). To abstract these differences, **libdebug** provides common utilities to resolve the stack trace (*backtrace*) of the running process (or thread).
+
+<div style="text-align: center;">
+    <img src="../../assets/backtrace_plot.png#only-light" loading="lazy" width="85%" />
+    <img src="../../assets/backtrace_plot_dark.png#only-dark" loading="lazy" width="85%" />
+</div>
+
+**libdebug**'s *backtrace* is structured like a LIFO stack, with the top-most value being the current instruction pointer. Subsequent values are the return addresses of the functions that were called to reach the current instruction pointer.
+
+!!! ABSTRACT "Backtrace usage example"
+    ```python
+    from libdebug import debugger
+
+    d = debugger("test_backtrace")
+    d.run()
+
+    # A few calls later...
+    [...]
+
+    current_ip = d.backtrace[0]
```

Comment:
This is using `backtrace` as a property, not as a function, whereas the example above shows `backtrace()[0]`. I sort of like this way more, so we might change it for the release.

Reply:
My bad

Reply:
Addressed in [288cb8b](https://github.com/libdebug/libdebug/commit/288cb8b5bd711382cc6116993dfb4c94cf498426)